examples/face_recognition_11/face_recognition_11/evaluator.py

```diff
@@ -0,0 +1,405 @@
+    else:
+        bbox_fte = True
+
+    # Stage 2: Keypoints
+
+    keypoint_fta = False
+    mse, nmse = None, None
+    Δ_nose, norm_Δ_nose = None, None
+    Δ_left_eye, norm_Δ_left_eye = None, None
+    Δ_right_eye, norm_Δ_right_eye = None, None
+    Δ_left_mouth, norm_Δ_left_mouth = None, None
+    Δ_right_mouth, norm_Δ_right_mouth = None, None
+    if inference.keypoints is not None:
+        normalization_factor = ground_truth.normalization_factor
+        Δ_nose, norm_Δ_nose = compute_distances(
+            ground_truth.keypoints.points[0], inference.keypoints.points[0], normalization_factor
+        )
+        Δ_left_eye, norm_Δ_left_eye = compute_distances(
+            ground_truth.keypoints.points[1], inference.keypoints.points[1], normalization_factor
+        )
+        Δ_right_eye, norm_Δ_right_eye = compute_distances(
+            ground_truth.keypoints.points[2], inference.keypoints.points[2], normalization_factor
+        )
+        Δ_left_mouth, norm_Δ_left_mouth = compute_distances(
+            ground_truth.keypoints.points[3], inference.keypoints.points[3], normalization_factor
+        )
+        Δ_right_mouth, norm_Δ_right_mouth = compute_distances(
+            ground_truth.keypoints.points[4], inference.keypoints.points[4], normalization_factor
+        )
+        distances = np.array([Δ_left_eye, Δ_right_eye, Δ_nose, Δ_left_mouth, Δ_right_mouth])
+        mse, nmse = calculate_mse_nmse(distances, normalization_factor)
+
+    if not bbox_fte and (inference.keypoints is None or nmse >= configuration.nmse_threshold):
+        keypoint_fta = True
+
+    # Stage 3: Recognition
+    pair_samples = list()
+    count_fnm, count_fm, count_tm, count_tnm = 0, 0, 0, 0
+    for i, (is_same, similarity) in enumerate(zip(ground_truth.matches, inference.similarities)):
+        is_match, is_false_match, is_false_non_match, failure_to_enroll = False, False, False, False
+        if similarity is None:
+            if is_same:
+                is_false_non_match = True
+            failure_to_enroll = True
+        elif similarity > threshold:  # match
+            if is_same:
+                is_match = True
+            else:
+                is_false_match = True
+        else:  # no match
+            if is_same:
+                is_false_non_match = True
+
+        pair_sample = PairSample(
+            locator=test_sample.pairs[i].locator if test_sample else "",
+            is_match=is_match,
+            is_false_match=is_false_match,
+            is_false_non_match=is_false_non_match,
+            failure_to_enroll=failure_to_enroll,
+            similarity=similarity,
+        )
+        pair_samples.append(pair_sample)
+        count_fnm += is_false_non_match
+        count_fm += is_false_match
+        count_tm += is_match
+        count_tnm += not is_false_non_match and not is_false_match and not is_match
+
+    return TestSampleMetrics(
+        pair_samples=pair_samples,
+        count_FNM=count_fnm,
```

Comment:
I think instead of incrementing in the loop if we can just compute them from there, it will be cleaner. `count_FNM = np.sum([pair.is_false_non_match for pair in pair_samples])`

Reply:
bump